tests/integration/test_package_managers.py

```diff
@@ -0,0 +1,47 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import pytest
+import os
+import utils
+import json
+from git import Repo
+
+# TODO: QUESTION - How we want to store configuration parts?
+CACHI2_IMAGE = "quay.io/containerbuildsystem/cachi2:latest"
+
+
+@pytest.mark.parametrize(
+    "env_package,env_name,should_fail,expected_rc,expected_output",
+    [
+        ("gomod_packages", "with_deps", False, 0,
+         "All dependencies fetched successfully"),
+    ],
+)
+def test_packages(env_package, env_name, should_fail, expected_rc, expected_output, tmpdir):
+    """
+    TODO: Rewrite docstring
+    """
+    env_data = utils.load_test_data(f"{env_package}.yaml")[env_name]
+    source_folder = utils.clone_repository(env_data['repo'], env_data['ref'], tmpdir)
+
+    #TODO: QUESTION - Pulling cachi2 image should be some before_scenario action maybe a fixture -> will be the same for all test cases
+    cachi2 = utils.ContainerImage(CACHI2_IMAGE)
```

Comment:
I'd like to have separation between test cases and the layer that runs cachito, can we have it as fixture, so test cases really cares only about specific test case and not pulling/running image?

Reply:
yeah +1

Reply:
+1